massa-network-worker/src/handshake_worker.rs

```diff
@@ -31,12 +31,67 @@ use massa_network_exports::{
 use massa_signature::KeyPair;
 use massa_time::MassaTime;
 use rand::{rngs::StdRng, RngCore, SeedableRng};
-use tokio::{task::JoinHandle, time::timeout};
-use tracing::debug;
+use std::collections::HashMap;
+use std::thread::{self, JoinHandle};
+use tokio::runtime::Handle;
+use tokio::time::timeout;
 
 /// Type alias for more readability
 pub type HandshakeReturnType = Result<(NodeId, ReadBinder, WriteBinder), NetworkError>;
 
+/// Start a thread, that will be responsible for running handshake workers,
+/// and sending the results back to the network worker.
+pub fn start_handshake_manager(
+    worker_rx: Receiver<(ConnectionId, HandshakeWorker)>,
+    connection_sender: Sender<(ConnectionId, HandshakeReturnType)>,
+    runtime_handle: Handle,
+) -> JoinHandle<()> {
+    thread::spawn(move || {
+        let mut task_handles = HashMap::new();
+        let (result_tx, result_rx) = bounded::<(ConnectionId, HandshakeReturnType)>(1);
```

Comment:
> use the type FuturesUnordered

It may be an option, but since the goal of the PR is to remove async where possible, I would rather use the present solution. 

> It's not possible that the branch L.54 from the select is triggered a second time before managing the first message

It's possible but it doesn't matter, since sending on the `result_tx` is inside a task spawned with `spawn_blocking`(therefore blocking on the send is not a problem).

Reply:
> It may be an option, but since the goal of the PR is to remove async where possible, I would rather use the present solution.

But your hashmap store async tasks so it's the same thing no ?

Reply:
And it will avoid the whole select logic and remove a layer of message sending which optimize a lot.

Reply:
> optimize a lot.

I doubt such a change would be noticeable, or even an improvement. We can optimize problems that are known to have a performance impact on the system, and avoid premature optimizations. 

> But your hashmap store async tasks so it's the same thing no ?

It's not the same thing, because the point of using [`FuturesUnordered`](https://docs.rs/futures/latest/futures/stream/struct.FuturesUnordered.html) is really to include it in a async `select`, and in this case the `select` is not async. 

In this situation, whenever a handshake completes, we want to both remove it from the hashmap, and send the result back to the network worker. When the handshake manager exits, we also want to abort any handshake that is still running. 

Also, the only async part is now reduced to the `new_handshake_worker.run().await`, while the surrounding code is using the native threading stuff--this fits in the goal of the PR.